### PR TITLE
test: Refactor test_cc_set_hostname.py and test_cc_ntp.py

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
test: Refactor test_cc_set_hostname.py and test_cc_ntp.py

* pytestify
* remove FilesystemMockingTestCase base class
* fix typing
* general cleanup

Any typing changes are to satisfy mypy.
```

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/unittests/config/test_cc_ntp.py [hunk 1]
```diff
@@ -457,7 +443,7 @@ def test_ntp_the_whole_package(self, m_sysd, m_select, m_which, m_subp):
             # skip the test if the distro is COS. As in COS, the default
             # config file is installed
             if distro == "cos":
-                return
+                continue
```
Comment (blackboxsw) [inline tests/unittests/config/test_cc_ntp.py:446]: good catch/fix

File: tests/unittests/config/test_cc_ntp.py [hunk 2]
```diff
@@ -609,11 +591,13 @@ def test_ntp_custom_client_overrides_installed_clients(
...
             # client is not installed
             m_which.return_value = None
-            mycloud = self._get_cloud(distro)
+            mycloud = get_cloud(distro)
             with mock.patch.object(
                 mycloud.distro, "install_packages"
-            ) as m_install:
-                cc_ntp.handle("notimportant", cfg, mycloud, None)
+            ) as m_install, mock.patch.object(
+                mycloud.distro, "manage_service"
```
Comment (blackboxsw) [inline tests/unittests/config/test_cc_ntp.py:598]: Good to plug these test leaks everywhere.

File: tests/unittests/config/test_cc_ntp.py [hunk 3]
```diff
@@ -695,43 +681,44 @@ def test_ntp_user_provided_config_template_only(
         }
         expected_merged_cfg = {
             "check_exe": "chronyd",
-            "confpath": "{tmpdir}/client.conf".format(tmpdir=self.new_root),
+            "confpath": "{tmpdir}/client.conf".format(tmpdir=tmpdir),
```
Comment (blackboxsw) [inline tests/unittests/config/test_cc_ntp.py:684]:
```suggestion
            "confpath": f"{tmpdir}/client.conf",
```

File: tests/unittests/config/test_cc_set_hostname.py [hunk 4]
```diff
@@ -156,195 +150,106 @@ def test_photon_hostname(self, m_subp):
...
-        cc = cloud.Cloud(ds, paths, {}, distro, None)
-        self.patchUtils(self.tmp)
-        cc_set_hostname.handle("cc_set_hostname", cfg, cc, [])
-        with self.assertRaises(FileNotFoundError):
-            util.load_text_file("/etc/hostname")
-
-    def test_create_hostname_file_false_rhel(self):
+        assert "blah" == contents.strip()
+
+    @pytest.mark.parametrize(
```
Comment (blackboxsw) [inline tests/unittests/config/test_cc_set_hostname.py:235]: I :heart: pytest parametrize

**Review by blackboxsw — APPROVED** [review]
LGTM, minor nits. Thank you @TheRealFalcon!

**Comment by TheRealFalcon** [comment]
@blackboxsw , I believe I have addressed your comments.

**Pull request merged**